cmd/modern/root/config/add-context.go

```diff
@@ -93,4 +94,23 @@ func (c *AddContext) run() {
 		{localizer.Sprintf("To start interactive query session"), "sqlcmd query"},
 		{localizer.Sprintf("To run a query"), "sqlcmd query \"SELECT @@version\""},
 	}, localizer.Sprintf("Current Context '%v'", context.Name))
+	c.LogTelemtry()
+}
+
+func (c *AddContext) LogTelemtry() {
+	eventName := "config-add-context"
+	properties := map[string]string{}
+	properties["Command"] = "config"
+	properties["SubCommand"] = "add-context"
+	if c.name != "" {
+		properties["name"] = "set"
+	}
```

Comment:
This function looks mostly like boilerplate that could be implemented in some common place and just have the command implementation pass the values not the property names for Command and SubCommand.

It's probably also useful to define a new type like

```Go
type CommandEvent map[string]string

func (e *CommandEvent) NewCommandEvent(command string, subCommand string) {
// Set Command and SubCommand properties on e
}

func (e *CommandEvent) SetProperty(p string, v string) *CommandEvent {
// Add the given property to the map and return e
}
```